lib/web_ui/lib/src/engine/keyboard_binding.dart

```diff
@@ -273,29 +304,14 @@ class KeyboardConverter {
            (metaDown ? _kDeadKeyMeta : 0);
   }
 
-  // Whether `event.key` should be considered a key name.
+  // Whether `event.key` is a key name, such as "Shift", or otherwise a
+  // character, such as "S" or "ж".
   //
-  // The `event.key` can either be a key name or the printable character. If the
-  // first character is an alphabet, it must be either 'A' to 'Z' ( and return
-  // true), or be a key name (and return false). Otherwise, return true.
-  static bool _eventKeyIsKeyname(String key) {
-    assert(key.isNotEmpty);
-    return isAlphabet(key.codeUnitAt(0)) && key.length > 1;
-  }
-
-  static int _characterToLogicalKey(String key) {
-    // Assume the length being <= 2 to be sufficient in all cases. If not,
-    // extend the algorithm.
-    assert(key.length <= 2);
-    int result = key.codeUnitAt(0) & 0xffff;
-    if (key.length == 2) {
-      result += key.codeUnitAt(1) << 16;
-    }
-    // Convert upper letters to lower letters
-    if (result >= _kCharUpperA && result <= _kCharUpperZ) {
-      result = result + _kCharLowerA - _kCharUpperA;
-    }
-    return result;
+  // A key name always has more than 1 code unit, and they are all alnums.
+  // Character keys, however, can also have more than 1 code unit: en-in
+  // maps KeyL to L̥/l̥. To resolve this, we check the second code unit.
+  static bool _eventKeyIsKeyName(String key) {
+    return key.length > 1 && key.codeUnitAt(0) < 0x7F && key.codeUnitAt(1) < 0x7F;
```

Comment:
Can't you just say `key.length > 2`? Are there any names that are only two characters?

Reply:
Yes, such as `Fn` and `F1`. And I'm not sure if there are characters of 3 or more code units.

Reply:
Ahh, OK.  LGTM then.